client/core/core.go

```diff
@@ -60,6 +60,12 @@ type dexConnection struct {
 
 	tradeMtx sync.RWMutex
 	trades   map[order.OrderID]*trackedTrade
+
+	// connected implies that the dex is connected.
+	//
+	// NOTE: This field is only used for notifications to the user, and
+	// should not be trusted in operations.
```

Comment:
Can we say why?  `wsConn.connected` is set when `dexConnection.connected` is set, although async (in a goroutine).  `connected=true` just means we think it is connected or rather we can treat it as connected until a send to the server could fail and the link will flip to `connected=false`, whereas `connected=false` means the link is definitely not connected and on reconnect it will be changed back to true.

I see no problem with modifying the client's behavior based on the best available information.